src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -270,6 +271,7 @@ public synchronized AmplitudeClient initialize(final Context context, final Stri
                     }
                     httpClient = new OkHttpClient();
                     deviceInfo = new DeviceInfo(context);
+                    deviceInfo.setLocationListening(this.locationListening);
```

Comment:
suggestion: you could add `locationListening` as an argument to `DeviceInfo` constructor since you already have it defined here